source/code/source/playfab/PlayFabWinHttpPlugin.cpp

```diff
@@ -192,6 +193,14 @@ namespace PlayFab
                     {
                         // Add HTTP headers
                         SetPredefinedHeaders(reqContainer, hRequest);
+                        if(!headerNoErrors)
```

Comment:
why is SetPredefinedHeaders returning void and relying on coupling with a member variable to report errors? this would be much more straight forward if SetPredefinedHeaders just returned the HRESULT

```
HRESULT hr = SetPredefinedHeaders(...);
if (FAILED(hr))
{
    ....
}
```

Reply:
Additionally, if you return an hresult from SetPredefinedHeaders you can just use the return value on the SetErrorInfo line below instead of calling GetLastError which isn't as clear IMO

Reply:
We are not changing any current method signatures. 

I stated in the updated commit comment that this function call is inherited by all other HTTPPlugins. This particular task only covers WinHttp issues. 

Yes I agree it's a better design to update the function signature to return the underlying HRESULT (trust me it looked WAY better to handle errors inside that function rather than relying on this flag), but since HttpPlugin is public and this function is virtual, this would be a breaking change to users.

---
In reply to: [412662871](https://github.com/PlayFab/XPlatCoreTemplate/pull/93#discussion_r412662871) [](ancestors = 412662871)

Reply:
oh I think I see. you've designed the API such that interested parties can inherit from the winhttpplugin. oof that's unfortunate.

Can you at least update the variable name to something easier to read than `headerNoErrors` that more clearly identifies the state as being coupled to SetPredefinedHeaders? e.g. `setPreDefinedHeadersFailure` or `setPredefinedHeadersResult` or something like that.

additionally, I think having SetPredefinedHeaders use a std::atomic<HRESULT> instead of std::atomic<bool> which requires calling GetLastError is still goodness

Reply:
This is not correct. I saw the function in the Windows plugin was set to virtual, but it really has no need of being that way, 

---
In reply to: [413170386](https://github.com/PlayFab/XPlatCoreTemplate/pull/93#discussion_r413170386) [](ancestors = 413170386,412662871)

Reply:
aren't all of the helper methods (and internal state) of the winhttp plugin marked protected? I think in theory that means a consumer could inherit from the winhttp plugin and take a dependency on the SetPredefinedHeaders method, no? I don't think that's a good idea and if it wasn't an intentionally designed use case maybe it's ok to break, but as is I think it's still possible for consumers to take weird dependencies on the winhttpplugin.

Reply:
I'm setting to won't fix this particular design decision to preserve whatever original state the code is currently in (focusing on a "keeping the lights on" approach here, rather than re-wiring)

---
In reply to: [413186169](https://github.com/PlayFab/XPlatCoreTemplate/pull/93#discussion_r413186169) [](ancestors = 413186169)